src/spells.cpp

```diff
@@ -236,7 +236,8 @@ uint32_t Spells::getInstantSpellCount(const Player* player) const
 
 InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
 {
-	auto& it = std::advance(instants.begin(), spellId);
+	auto& it = instants.begin();
+	std::advance(it, spellId);
```

Comment:
I just [read here](http://en.cppreference.com/w/cpp/iterator/advance) that:

> The behavior is undefined if the specified sequence of increments or decrements would require that a non-incrementable iterator (such as the past-the-end iterator) is incremented, or that a non-decrementable iterator (such as the front iterator or the singular iterator) is decremented.

So it's probably smart to use `std::next(test.begin(), std::max(n, test.size()))`.